src/compiler/scala/tools/nsc/typechecker/RefChecks.scala

```diff
@@ -1661,18 +1661,15 @@ abstract class RefChecks extends Transform {
         case TypeApply(fun, targs) =>
           isClassTypeAccessible(fun)
         case Select(module, apply) =>
-          ( // scala/bug#4859 `CaseClass1().InnerCaseClass2()` must not be rewritten to `new InnerCaseClass2()`;
-            //          {expr; Outer}.Inner() must not be rewritten to `new Outer.Inner()`.
-            treeInfo.isQualifierSafeToElide(module) &&
-              // scala/bug#5626 Classes in refinement types cannot be constructed with `new`. In this case,
-              // the companion class is actually not a ClassSymbol, but a reference to an abstract type.
-              module.symbol.companionClass.isClass
-            )
+          // scala/bug#4859 `CaseClass1().InnerCaseClass2()` must not be rewritten to `new InnerCaseClass2()`;
+          //          {expr; Outer}.Inner() must not be rewritten to `new Outer.Inner()`.
+          treeInfo.isQualifierSafeToElide(module) &&
+          // scala/bug#5626 Classes in refinement types cannot be constructed with `new`.
+          !module.exists { case t @ Select(_, _) => t.symbol != null && t.symbol.isStructuralRefinementMember case _ => false }
```

Comment:
😵‍💫 

> It's an interesting breakage.

What broke? Using `ClassForCaseCompanionAttachment` would seem more straightforward.

Reply:
The t5626 broke. IIRC, the old comment did not hold, the companion was `NoSymbol` and trivially not class, not the "abstract type". (I think the code was changing at the time, so I don't know when that test no longer worked?) At https://scala-ci.typesafe.com/job/scala-2.13.x-validate-main/17861/console
```
t5626.scala:10: error: Internal error: unable to find the outer accessor symbol of class C
  val foo = blob.Foo()
      ^
t5626.scala:11: error: Internal error: unable to find the outer accessor symbol of class C
  val bar = blub.Foo()
      ^
2 errors
```
I intended to brush up before the final exam.

Reply:
Ah I see 👍